### PR TITLE
Improve feedbacks on read-only remote filesystems

### LINKED ISSUE #2081: Preserve time attribute in created ZIP archive file
**Is your feature request related to a problem? Please describe.**
When creating a ZIP archive file, the date and time attributes do not get preserved.

**Describe the solution you'd like**

The date and time attributes should be preserved inside the ZIP file. This is especially useful when wanting to upload many small files to an FTP server with no [MFMT support](https://github.com/TeamAmaze/AmazeFileManager/issues/1570#issuecomment-721959779).

 Obviously, this requires no root access because it is not on file-system or block level.

### LINKED ISSUE #2085: R/O access user experience
**Describe the bug**
Currently the user experience is not good in case SMB client has R/O access. User can't see it anywhere on screen, and as he performs any R/W operation, there's no 'Failed' or 'Not permitted' callback shown to the user. It just silently closes with following log.

**Expected behavior**
Can extend to SFTP client as well. Better maintain connection permission state and show somewhere in UI all the time so user doesn't perform R/W operations.

**Additional context**

```
2020-11-13 12:55:17.852 29613-31247/com.amaze.filemanager.debug W/System.err: jcifs.smb.SmbAuthException: Access is denied.
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.checkStatus(SmbTransportImpl.java:1376)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.checkStatus(SmbTransportImpl.java:1573)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.sendrecv(SmbTransportImpl.java:1023)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.send(SmbTransportImpl.java:1539)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbSessionImpl.send(SmbSessionImpl.java:409)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:472)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:404)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:318)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:298)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbFile.renameTo(SmbFile.java:1353)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbFile.renameTo(SmbFile.java:1303)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at com.amaze.filemanager.filesystem.Operations$3.doInBackground(Operations.java:396)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at com.amaze.filemanager.filesystem.Operations$3.doInBackground(Operations.java:369)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.lang.Thread.run(Thread.java:919)
```

### PR DESCRIPTION
**Pull request by TranceLove**
#### Issue tracker   
Fixes will automatically close the related issue

Fixes #2085

#### Release  
Addresses release/3.5
  
#### Test cases
- [x] Covered
  
#### Manual testing
- [ ] Done  
  
If yes,  
- Device:
- OS:

#### Build tasks success  
Successfully running following tasks on local 
- [ ] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

#### Additional Info
- Prompt user with dialog on unable to delete/move/rename files for SMB/SFTP files
- Add test case for SMB/SSH rename and delete

### CODE REVIEW

**Review by EmmanuelMess — APPROVED** [review]

**Comment by EmmanuelMess** [comment]
Will test

**Review by VishalNehra — DISMISSED** [review]

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/DeleteTask.java [hunk 1]
```diff
@@ -80,111 +87,57 @@ protected void onProgressUpdate(String... values) {
...
-          e.printStackTrace();
-          wasDeleted = false;
-          break;
-        }
+    for (HybridFileParcelable file : files) {
+      try {
+        wasDeleted = doDeleteFile(file);
+        if (!wasDeleted) break;
+      } catch (Exception e) {
+        return new AsyncTaskResult<>(e);
```
[the remark this answers is not in the document]
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/DeleteTask.java:103]: The exception was for Toast or other UI notifications to feedback user on what exactly happened.

Now we are simply broadcast Intent for the dialog, yes, `AsyncTaskResult` with exception can be changed back to `AsyncTaskResult(false)`.

[the remark this answers is not in the document]
Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/DeleteTask.java:103]: The return type for doInBackground is defined as `AsyncTask<Boolean>` when extending AsyncTask on the top, how does it accept `AsyncTask<Exception>` in return is what I'm not getting.

**Comment by EmmanuelMess** [comment]
I am losing the time attribute on zips:
Screenshot_1606317799Screenshot_1606317803

**Comment by TranceLove** [comment]
@EmmanuelMess I suppose this is tracked by #2081?

**Comment by EmmanuelMess** [comment]
> @EmmanuelMess I suppose this is tracked by #2081?

This PR says that it fixes 2081 though.

**Comment by TranceLove** [comment]
@EmmanuelMess Oops, my bad. Force push to happen soon.

**Review by VishalNehra — APPROVED** [review]

**Review by EmmanuelMess — APPROVED** [review]

**Pull request merged**